### PR TITLE
Fix textfile ambiguous timestamps and different storage timezones

### PR DESCRIPTION
**Pull request by rschlussel**
## Description
For TEXTFILE tables, when structural types (array, map, row) contain timestamp columns, we weren't converting to the hive storage time zone as we were doing for primitive types. Instead the timestamps would be interpreted in the JVM timezone

The conversion to hive storage time zone had a secondary benefit (even when the timezones were the same) of fixing the handling of ambiguous timestamps. Ambiguous timestamps are local times that can have more than one unixtime representation. It happens commonly during the fall DST conversion where the hour from 1-2am repeats. Generally in Presto we use the earlier of thetwo possible times when the unixtime is ambiguous. However, the hive library we use for parsing textfiles uses the later time. The code for adjusting the time based on the hive storage time zone has a secondary benefit of correcting ambiguous timestamps to the earlier unixtime representation.

This change fixes those two issues for the structural type code path.

## Motivation and Context
The motivation for this change is to provide consistent results for ambiguous timestamps regardless of where it is used.  We also want to fix incorrectly not using the storage time zone for timestamps in structural types.

## Impact
Ambiguous timestamps inside structural types in textfiles will now be interpreted in the earliest possible unixtime.
The hive.time-zone property will now be respected for timestamps inside structural types in textfiles

## Test Plan
new tests

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Hive Connector Changes
* Fix timestamps inside array, map, or row types for tables using TEXTFILE format to respect the ``hive.time-zone property``.
* Fix interpretation of ambiguous timestamps inside array, map, or row types for tables using TEXTFILE format to interpret the timestamps as the earliest possible unixtime for consistency with the rest of Presto.

### CODE REVIEW

**Review by spershin — APPROVED** [review]
Thank you for looking into this and fixing it @rschlussel !

Looks good, need a committer's approval too.

File: presto-hive/src/main/java/com/facebook/presto/hive/util/SerDeUtils.java
Comment (rschlussel) [inline presto-hive/src/main/java/com/facebook/presto/hive/util/SerDeUtils.java:297]: This is the part with the change. Logic is copied from https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java#L320-L333

**Review by zacw7 — APPROVED** [review]
Thanks for the fix!!

**Review by elharo — APPROVED** [review]

**Pull request merged**
